compiler/daml-lf-tools/src/DA/Daml/LF/Simplifier.hs

```diff
@@ -493,6 +489,12 @@ isWorthLifting = \case
     EEnumCon _ _ -> False
     ENil _ -> False
     ENone _ -> False
+    ETmLam _ _ -> False
```

Comment:
I'm not convinced we shouldn't lift lambdas at all. In a chain of lambdas, we should still lift the outermost lambda when possible. For instance, in
```haskell
f x y z =
  let g a b = a+b
  in f x (f y z)
```
I would expect the `g` to be lifted to the top-level.

We absolutely need tests for the constant lifter. We could either do very specific unit test with hand crafted ASTs or we could use the `jq` based approach.